src/envoy/http/authn/authenticator_base.cc

```diff
@@ -71,17 +70,49 @@ void AuthenticatorBase::validateX509(
     done_callback(&payload, false);
   }
 
-  // TODO (lei-tang): Adding other attributes (i.e ip) to payload if desire.
+  // TODO (lei-tang): Adding other attributes (i.e ip) to payload if needed.
   done_callback(&payload, true);
 }
 
+// TODO (lei-tang): Extracting more attributes/claims to payload if needed.
+void extractJwtPayload(
+    ::google::protobuf::Map< ::std::string, ::std::string>& jwt_payload,
+    Payload* payload) {
+  // Extract attributes from JWT payload
+  // Extract user
+  if (jwt_payload.count("iss") > 0 && jwt_payload.count("sub") > 0) {
+    payload->mutable_jwt()->set_user(jwt_payload["iss"] + "/" +
+                                     jwt_payload["sub"]);
+  }
+  // Extract audience
+  if (jwt_payload.count("aud") > 0) {
```

Comment:
Do you know how multiple audiences are stored jwt output? Do we need to parse this? If if not supported, add a note for that here. + @qiwzhang

Reply:
Added a note that the aud field is taken from the jwt-auth produced claim as is. Wayne knows more about how multiple audiences are stored/formatted by jwt-auth. I think Istio authn does not need to parse the aud field because it is jwt-auth that validates the JWT audiences and other JWT attributes whereas Istio authn simply passes these attributes to next consumers.

Reply:
Added the handling of the case when aud may be an array of strings or a single string